server/homebrew.api.js

```diff
@@ -6,13 +6,59 @@ const GoogleActions = require('./googleActions.js');
 const Markdown = require('../shared/naturalcrit/markdown.js');
 const yaml = require('js-yaml');
 const asyncHandler = require('express-async-handler');
+const { nanoid } = require('nanoid');
 
 // const getTopBrews = (cb) => {
 // 	HomebrewModel.find().sort({ views: -1 }).limit(5).exec(function(err, brews) {
 // 		cb(brews);
 // 	});
 // };
 
+const findBrew = async (brewId, accessType, googleObjectId)=>{
+	let id = brewId;
+	let googleId = googleObjectId;
+	if(id.length > 12) {
+		googleId = id.slice(0, -12);
+		id = id.slice(-12);
+	}
+	let brew = await HomebrewModel.get(accessType === 'edit' ? { editId: id } : { shareId: id })
+		.catch((err)=>{
+			if(googleId) {
+				console.warn(`Unable to find document stub for ${accessType}Id ${id}`);
+			} else {
+				console.warn(err);
+			}
+		});
+
+	let googleBrew;
+	if(googleId || brew?.googleId) {
+		googleBrew = await GoogleActions.getGoogleBrew(googleId || brew?.googleId, id, accessType)
+			.catch((err)=>{
+				console.warn(err);
+			});
+		if(!brew && !googleBrew) throw 'Brew not found in database or Google Drive';
+		brew = brew ? _.merge(brew, googleBrew) : googleBrew;
+	} else if(!brew) {
+		throw 'Brew not found in database';
+	}
+
+	return brew;
+};
+
+const getBrew = (accessType)=>{
+	return async (req, res, next)=>{
+		const { brew } = req;
+
+		if(!brew) {
+			const id = req.params.id, googleId = req.body?.googleId;
+			const found = await findBrew(id, accessType, googleId);
+			req.brew = accessType !== 'edit' && found.toObject ? found.toObject() : found;
+		}
+
+		!!next ? next() : undefined;
```

Comment:
I don't think we need to worry about `next()` not existing. Express.js should know how to handle it if we just call `next()`.